### PR TITLE
Add package build tests

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Add tests for built package, for freshly built package functionality check (#1479).

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -116,6 +116,47 @@ jobs:
         with:
           parallel-finished: true
 
+  package-test:
```
Comment (arkid15r) [inline .github/workflows/ci-cd.yml:119]: Let's make the naming consistent: `build` + `test-build` or `package` + `test-package`

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -116,6 +116,47 @@ jobs:
...
+      - name: Get package artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: dist
+          path: dist
+
+      - name: Install package and dependencies
+        shell: bash
+        run: |
+          mv holidays _holidays
```
Comment (arkid15r) [inline .github/workflows/ci-cd.yml:151]: Can we just remove `holidays` directory?

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -116,6 +116,47 @@ jobs:
...
+          PACKAGENAME=`ls dist/*.whl`
+          python -m pip install $PACKAGENAME
```
Comment (arkid15r) [inline .github/workflows/ci-cd.yml:153]:
```suggestion
          python -m pip install `ls dist/*.whl`
```

File: .github/workflows/ci-cd.yml [hunk 4]
```diff
@@ -116,6 +116,47 @@ jobs:
...
+          python -m pip install --upgrade --requirement requirements/tests.txt
+
+      - name: Run tests
+        run: |
+          pytest --dist loadscope --numprocesses auto tests/countries tests/financial
+
```
Comment (arkid15r) [inline .github/workflows/ci-cd.yml:159]: Do we want to add similar test for `ls dist/*.tar.gz` ?

Comment (KJhellico) [reply .github/workflows/ci-cd.yml:159]: Yes, for full coverage, it's probably needed.

**Review by arkid15r — APPROVED** [review]
LGTM

File: .github/workflows/ci-cd.yml [hunk 5]
```diff
@@ -116,6 +116,46 @@ jobs:
...
+          path: dist
+
+      - name: Run tests
+        shell: bash
+        run: |
+          rm -rf holidays
+          python -m pip install --upgrade --requirement requirements/tests.txt
+          python -m pip install `ls dist/*.whl`
+          pytest --dist loadscope --numprocesses auto tests/countries tests/financial
+          python -m pip uninstall -y holidays python-dateutil six
```
[the remark this answers is not in the document]
Comment (KJhellico) [reply .github/workflows/ci-cd.yml:155]: It's install as `python-dateutil` dependency.

**Pull request merged**